public/js/views/purchase.jsx

```diff
@@ -33,49 +33,48 @@ export default class Purchase extends Component {
 
   selectData(state) {
     return {
-      purchase: state.purchase,
+      transaction: state.transaction,
       user: state.user,
     };
   }
 
-  renderChild(result) {
+  renderChild(connector) {
     var props = this.props;
     var BraintreeToken = props.BraintreeToken;
     var CompletePayment = props.CompletePayment;
     var CardListing = props.CardListing;
     var AddSubscription = props.AddSubscription;
 
-    if (result.purchase.completed) {
+    if (connector.transaction.completed) {
       return (
         <CompletePayment
           productId={props.productId}
           userEmail={props.user.email} />
       );
-    } else if (result.purchase.payMethods &&
-               result.purchase.payMethods.length > 0) {
+    } else if (connector.transaction.availablePayMethods.length > 0) {
       console.log('rendering card listing');
       return (
         <CardListing
           productId={props.productId}
-          payMethods={result.user.payMethods}
-          {...bindActionCreators(purchaseActions, result.dispatch)}
+          payMethods={connector.transaction.availablePayMethods}
```

Comment:
Yeah, exactly. It suddenly dawned on me that we need to make a distinction here that they are only the *available* pay methods. I think this makes the logic read a lot better because nearby areas refer to `user.payMethods` for different purposes.